FirebaseAuth/Sources/Swift/Backend/AuthBackend.swift

```diff
@@ -122,8 +120,9 @@ class AuthBackend {
        languageCode.count > 0 {
       request.setValue(languageCode, forHTTPHeaderField: "X-Firebase-Locale")
     }
-    if let appCheck = requestConfiguration.appCheck {
-      let tokenResult = await appCheck.getToken(forcingRefresh: false)
+    // Wait for the async header values.
+    await request.setValue(heartbeatsHeaderValue, forHTTPHeaderField: "X-Firebase-Client")
+    await appCheckTokenHeaderValue.map { tokenResult in
```

Comment:
Why the `map` instead of the original code?

Reply:
using `appCheckTokenHeaderValue` needs an `await` attached to it because it was assigned with `async let`. 

My first instinct was to try:
```swift
if await let tokenResult = appCheckTokenHeaderValue { ... }
```
But that didn't work, so I did the `map` which expresses the same idea.

Experimenting with it again now, you could also do:
```suggestion
    if let tokenResult = await appCheckTokenHeaderValue {
```
Do you have a preference?

Reply:
The second version is more readable to me.  Thanks